### PR TITLE
Create gomis-senegalese-medicine-thesis.csl

### PR DESCRIPTION
**Pull request by drgomis**
Vancouver style with some changes in order to create the right style for thesis and dissertations from Faculté de Médecine et d'Odontologie of Université Cheikh Anta Diop de Dakar. The main change is the display of the references in alphabetical order. We use the veterinary-pathololy.csl files to start this work.

### CODE REVIEW

**Comment by drgomis** [comment]
Vancouver style with some changes in order to create the right style for thesis and dissertations from Faculté de Médecine et d'Odontologie of Université Cheikh Anta Diop de Dakar. The main change is the display of the references in alphabetical order. We use the veterinary-pathololy.csl files to start this work.

File: universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl [hunk 1]
```diff
@@ -0,0 +1,316 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" page-range-format="expanded" demote-non-dropping-particle="sort-only" default-locale="fr-FR">
+  <info>
+    <title>Université Cheikh Anta Diop - Faculté de Médecine de Pharmacie et d'Odontologie</title>
```
Comment (adam3smith) [inline universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl:4]:
```suggestion
    <title>Université Cheikh Anta Diop - Faculté de Médecine de Pharmacie et d'Odontologie (Français)</title>
```

**Review by adam3smith — CHANGES_REQUESTED** [review]

**Comment by drgomis** [comment]
Thank you @adam3smith.

**Review by adam3smith — COMMENTED** [review]
Some more questions

File: universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    <terms>
+      <term name="collection-editor" form="long">
+        <single>editor</single>
+        <multiple>editors</multiple>
+      </term>
+      <term name="presented at">presented at</term>
+      <term name="available at">available from</term>
+      <term name="section" form="short">sect.</term>
+    </terms>
+  </locale>
```
[the remark this answers is not in the document]
Comment (drgomis) [reply universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl:35]: Hello @adam3smith,
What, I have to change in this section?

File: universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+  <macro name="report-details">
+    <choose>
+      <if type="report">
+        <text variable="number" form="short" strip-periods="false" quotes="false" font-variant="normal" font-weight="normal" text-decoration="none" vertical-align="baseline" prefix=". Rapport N°"/>
+      </if>
+    </choose>
+  </macro>
+  <citation collapse="citation-number">
+    <sort>
+      <key variable="author"/>
```
[the remark this answers is not in the document]
Comment (drgomis) [reply universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl:302]: Yes, it is the style of our faculty.

File: universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+    </sort>
+    <layout font-weight="normal" vertical-align="baseline" delimiter=", " prefix="[" suffix="]">
+      <text variable="citation-number" strip-periods="false" font-style="normal" font-variant="normal" font-weight="normal" text-decoration="none" vertical-align="baseline"/>
+    </layout>
+  </citation>
+  <bibliography et-al-min="7" et-al-use-first="3" second-field-align="flush">
```
[the remark this answers is not in the document]
Comment (drgomis) [reply universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl:308]: Thank you for this remark. It's et-al-use-first="6". I will change it.

File: universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl (hunk 3 above)
[the remark this answers is not in the document]
Comment (adam3smith) [reply universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl:286]: I understand you want alphabetical sorting in the bibliography, but do you really want [3, 1] or even [2,5,1] in text. I find that hard to believe

**Comment by adam3smith** [comment]
Your link to documentation doesn't actually go to documentation for the style - is that available somewhere? That would also help with answering the remaining question on sorting

File: universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl (hunk 3 above)
[the remark this answers is not in the document]
Comment (drgomis) [reply universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl:286]: No, in the text I want to put it in [1, 3] or [1, 2, 5], and if the numbers follow each other in [5-10].

**Comment by drgomis** [comment]
**> Your link to documentation doesn't actually go to documentation for the style - is that available somewhere? That would also help with answering the remaining question on sorting**

You can see, what I really want in these two documents in the attached files. The documentation isn't online yet because I just finished writing it.

[Modele automatisation des citations et references selon le fichier csl UCAD-FMPO. Gomis.pdf](https://github.com/citation-style-language/styles/files/7798237/Modele.automatisation.des.citations.et.references.selon.le.fichier.csl.UCAD-FMPO.Gomis.pdf)
[Proposition Guide de références bibliographiques theses memoires UCAD-FMPO. Gomis.pdf](https://github.com/citation-style-language/styles/files/7798238/Proposition.Guide.de.references.bibliographiques.theses.memoires.UCAD-FMPO.Gomis.pdf)

File: universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl (hunk 3 above)
[the remark this answers is not in the document]
Comment (drgomis) [reply universite-cheikh-anta-diop-faculte-de-medecine-de-pharmacie-et-dodontologie.csl:286]: I understand. I will change it right now to  "sorted in the text by citation-number".

**Comment by adam3smith** [comment]
(don't worry about the test failures; they're not because of you)

**Comment by drgomis** [comment]
Ok, I see. I think there is a problem with the csl validator.
So, thank you for the information.

**Comment by drgomis** [comment]
@adam3smith, when can I try again for the style test?

**Comment by drgomis** [comment]
Thank you @adam3smith.

**Comment by drgomis** [comment]
Thank you @adam3smith. I want to know how long it will take for the style to be uploaded in the Zotero repository?

**Comment by adam3smith** [comment]
Typically within an hour. It's available now

**Comment by drgomis** [comment]
I see. Thank you again.

**Pull request merged**